service/src/test/java/cz/muni/fi/pa165/mushrooms/service/facade/MushroomFacadeImplTest.java

```diff
@@ -0,0 +1,139 @@
+package cz.muni.fi.pa165.mushrooms.service.facade;
+
+import cz.muni.fi.pa165.mushrooms.dto.MushroomDTO;
+import cz.muni.fi.pa165.mushrooms.entity.Mushroom;
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+import cz.muni.fi.pa165.mushrooms.facade.MushroomFacade;
+import cz.muni.fi.pa165.mushrooms.service.BeanMappingService;
+import cz.muni.fi.pa165.mushrooms.service.MushroomService;
+import cz.muni.fi.pa165.mushrooms.service.config.ServiceConfiguration;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+import javax.inject.Inject;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for facade functionality without mock - using the real service implementation.
+ *
+ * @author Lindar84
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+public class MushroomFacadeImplTest extends AbstractTransactionalJUnit4SpringContextTests {
+
+    @Inject
+    MushroomService service;
+    @Inject
+    MushroomFacade facade;
+    @Inject
+    private BeanMappingService mapping;
+
+    private Mushroom mushroom1;
+    private Mushroom mushroom2;
+
+    private MushroomDTO mushroom1DTO;
+    private MushroomDTO mushroom2DTO;
+
+    private static Mushroom createMushroom(String name, MushroomType type, String from, String to) {
```

Comment:
move to TestUtils

Reply:
I change TestUtils to public because of problems with access